File: MergeToSurvive/src/com/vgdc/merge/entities/controllers/ProjectileController.java
package com.vgdc.merge.entities.controllers;

import com.vgdc.merge.entities.Entity;
import com.vgdc.merge.entities.Platform;

public class ProjectileController extends AbilityController{
	
	public ProjectileController(float timeAlive)
	{
		this.timeAlive = timeAlive;
	}
	
	private float timeAlive;
	
	@Override
	public void setEntity(Entity entity)
	{
		super.setEntity(entity);
	}

	@Override
	public Controller copy() {
		ProjectileController controller = new ProjectileController(timeAlive);
		return controller;
	}

	@Override
	public void onUpdate(float delta) {
		timeAlive-=delta;
		if(timeAlive<=0)
		{
			onDeath();
		}
	}
	
	@Override
	public void onPlatformCollision(Platform platform){
		System.out.println("Collision!");
		onDeath();
	}

	@Override
	public void onEntityCollision(Entity entity){
		if(entity.getTeam()!=getEntity().getTeam())
			onDeath();
	}

}

File: MergeToSurvive-android/assets/test/scripts/ProjectileController.py
from AbilityController import AbilityController
from com.vgdc.merge.entities import Entity
from com.vgdc.merge.entities import Platform

class ProjectileController(AbilityController):
	
	def __init__(self, time = 0):
		self.timeAlive = time
		
	def copy(self):
		controller = ProjectileController(self.timeAlive)
		return controller
		
	def onUpdate(self, delta):
		self.timeAlive-=delta
		if self.timeAlive<=0:
			print("Death by Time-Out!")
			if not self.getEntity().isDead():
				self.onDeath()
				
	def onPlatformCollision(self, platform):
		if not self.getEntity().isDead():
			print("Platform Collision!")
			self.onDeath()	
			
	def onEntityCollision(self, entity):
		if entity.getTeam() != self.getEntity().getTeam():
			print("Entity Collision!")
			if not self.getEntity().isDead():
				self.onDeath()